commons/com.b2international.index/src/com/b2international/index/revision/RevisionConflictProcessor.java

```diff
@@ -184,13 +185,13 @@ protected RevisionPropertyDiff handleCollectionConflict(DocumentMapping mapping,
 				// TODO nested item conflict resolution???
 				// create a big union of all changes, treat collection as set if needed
 				// this eliminates most of the unnecessary conflicts between two collection properties
-				sourceDiff.getAddedItems().forEach(oldArray::add);
-				Set<JsonNode> newSourceItems = Set.copyOf(sourceDiff.getAddedItems());
-				targetDiff.getAddedItems().forEach(newTargetItem -> {
-					if (!isSet || !newSourceItems.contains(newTargetItem)) {
-						oldArray.add(newTargetItem);
-					}
-				});
+				Set<JsonNode> oldArraySet = Sets.newHashSet(oldArray);
+				Streams.concat(sourceDiff.getAddedItems().stream(), targetDiff.getAddedItems().stream())
+					.forEach(newItem -> {
+						if (!isSet || !oldArraySet.contains(newItem)) {
```

Comment:
`oldArraySet.add(newItem)` would also update the backing Set as updates are streamed.

Reply:
True, updated, thanks!